backend/src/main/kotlin/com/woowahan/campus/announcement/feature/GetAllAnnouncement.kt

```diff
@@ -0,0 +1,70 @@
+package com.woowahan.campus.announcement.feature
+
+import com.woowahan.campus.announcement.domain.Announcement
+import com.woowahan.campus.announcement.domain.AnnouncementRepository
+import openapi.api.GetAllAnnouncementApi
+import openapi.model.AnnouncementInfoResponse
+import openapi.model.AnnouncementsInfoByCursorResponse
+import openapi.model.AnnouncementsInfoByOffsetResponse
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Slice
+import org.springframework.http.ResponseEntity
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@Transactional(readOnly = true)
+class GetAllAnnouncement(private val announcementRepository: AnnouncementRepository) : GetAllAnnouncementApi {
+    //TODO: 패스워드 레포지터리 및 디코더 구현
+    override fun findAllAnnouncementByOffset(
+        authorization: String,
+        page: Int,
+        size: Int,
+    ): ResponseEntity<AnnouncementsInfoByOffsetResponse> {
+        val announcements: Page<Announcement> = announcementRepository.findAll(PageRequest.of(page, size))
+
+        return ResponseEntity.ok(
+            AnnouncementsInfoByOffsetResponse(
+                announcements.get().map(::toAnnouncementPageResponses).toList(),
```

Comment:
요거 코틀린 ` map { } ` 으로 안되나요?

Reply:
```kotlin
announcements.get().map{ toAnnouncementPageResponses(it) }.toList(),
```

위에 처럼 활용할 수 있는데 `::` 을 사용해서 함수 레퍼런스를 호출하는게 더 깔끔해보이더리구요
박스터 생각은 어떠신가요?

Reply:
저렇게하면 toList를 안써도 되지않나요..?
```suggestion
                announcements.get().map{ toAnnouncementPageResponses(it) },
```

Reply:
<img width="1066" alt="image" src="https://github.com/Songusika/woowa-announcement-attendance/assets/74398096/b9e88ebf-a53d-4bc4-ace0-7588c8f9645e">

map 메서드가 Stream 을 반환하더라구요!